src/Core/RunFactory.php

```diff
@@ -40,6 +40,7 @@ public function fromSpider(SpiderInterface $spider, ?Overrides $overrides = null
 
         return new Run(
             $spider->getInitialRequests(),
+            $spider::class,
```

Comment:
We should also add a test to `RunFactoryTest.php` to check that the spider's class name gets passed as the default namespace.